File: widgets/heading.lua
local _context = getContext()
local _painter = _context:getPainter()
local _style = _painter:getStyle()

exports('heading', function(text)
	local headingStyle = _style.heading

	_painter:setText(text)
	_painter:setTextOpts(headingStyle.text.font, headingStyle.text.scale)

	local w = _context:getWidgetWidth() or _painter:calculateTextWidth()
	local h = headingStyle.height

	_context:beginDraw(w, h)

	_painter:setColor(_style.color.primary)
	_painter:drawText(headingStyle.text.offset)

	_context:endDraw()
end)

File: src/widgets/heading.ts
import { getCurrentContext } from '../../index';
import { Color } from '../common/types';

export function declareExport(): void {
	globalThis.exports('heading', function (text: string | undefined): void {
		const context = getCurrentContext();
		const painter = context.getPainter();
		const style = painter.getStyle();

		const headingStyle = style.heading;

		painter.setText(text);
		painter.setTextOpts(headingStyle.text.font, headingStyle.text.scale);

		const w = context.getWidgetWidth() ?? painter.calculateTextWidth();
		const h = headingStyle.height;

		context.beginDraw(w, h);

		painter.setColor(style.getProperty<Color>('heading', 'color'));
		painter.drawText(headingStyle.text.offset);

		context.endDraw();
	});
}
